### PR TITLE
apollo-runtime: Make a few classes internal 

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/internal/dispatchers.kt [hunk 1]
```diff
@@ -2,14 +2,14 @@ package com.apollographql.apollo3.internal
...
-expect fun defaultDispatcher(requested: CoroutineDispatcher?): CoroutineDispatcher
+internal expect fun defaultDispatcher(requested: CoroutineDispatcher?): CoroutineDispatcher
 
 /**
  * A mutex implementation for code that runs in the defaultDispatcher
  *
  * On native, this is a no-op because the defaultDispatcher is always the main loop
  */
-expect class DefaultMutex(): Mutex
+internal expect class DefaultMutex() : Mutex
```
Comment (BoD) [inline apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/internal/dispatchers.kt:12]: By the way (probably silly question) - why don't we use `kotlinx.coroutines.sync.Mutex`?

Comment (martinbonnin) [reply apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/internal/dispatchers.kt:12]: I think initially we used the `DefaultMutex` from non-suspending places but that doesn't seem to be the case anymore. Feel free to change to the coroutines version.

**Review by martinbonnin — APPROVED** [review]
Nice 🤩 ! Love it when the api size decreases !

File: apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/internal/dispatchers.kt (hunk 1 above)
Comment (BoD) [reply apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/internal/dispatchers.kt:12]: 👍 I will have a look!

**Pull request merged**